spring-session-data-mongodb/src/main/java/org/springframework/session/data/mongo/MongoIndexedSessionRepository.java

```diff
@@ -178,8 +181,16 @@ private void publishEvent(ApplicationEvent event) {
 		}
 	}
 
-	public void setMaxInactiveIntervalInSeconds(final Integer maxInactiveIntervalInSeconds) {
-		this.maxInactiveIntervalInSeconds = maxInactiveIntervalInSeconds;
+	/**
+	 * Set the maximum inactive interval in seconds between requests before newly created
+	 * sessions will be invalidated. A negative time indicates that the session will never
+	 * time out. The default is 30 minutes.
+	 * @param defaultMaxInactiveInterval the default maxInactiveInterval
+	 */
+	public void setDefaultMaxInactiveInterval(Duration defaultMaxInactiveInterval) {
+		org.springframework.util.Assert.notNull(defaultMaxInactiveInterval,
```

Comment:
I see that Mongo's Assert.requireNonNull is already used which is why this must be fully qualified. This usage is inconsistent, but it brings the question of why Spring's Assert.notNull isn't used in the other case. Perhaps we should log a ticket to align this?

Reply:
Along the lines of https://github.com/spring-projects/spring-session/pull/2169#discussion_r980348567, IMO we need to do a more general alignment of MongoDB session repository implementations. I'll open an issue to track that.